File: checkers/228/main.cpp
#include <checkers/testlib.h>
#include <iostream>
#include <fstream>
#include <stdlib.h>
#include <vector>
#include <iomanip>
#include <sstream>

using namespace NTestlib;

#define ld long double
#define eps 0.001
ld f1(ld x, ld y)
{
    return x * x + y * y - 0.1 * std::abs(1 - x) - 0.1 * std::abs(1 - y);
}
ld f2(ld x, ld y)
{
    return 20 * std::abs(x - 50) * std::abs(y - 25) + 10 * (std::abs(x - 10) * std::abs(y - 10) + std::abs(x - 50));
}
int main(int argc, char *argv[])
{
    InitInteractor(argc, argv);
    int f = File.ReadInt();
    int k = 0;
    while (Out.HasInput())
    {
        k++;
        if (k > 100000)
            QuitWith(WA, "Too many requests");
        char requestType = Out.ReadChar();
        ld x = Out.ReadDouble();
        ld y = Out.ReadDouble();
        switch (requestType)
        {
        case '!':
        {
            if (f)
                if (std::abs(x + 0.05) < eps && abs(y + 0.05) < eps)
                    QuitWith(AC, "Full solution");
                else
                {
                    std::stringstream msg;
                    msg << "Incorrect output: " << x << " " << y;
                    QuitWith(WA, msg.str());
                }
            else if (std::abs(50 - x) < eps && std::abs(10 - y) < eps)
                QuitWith(AC, "Full solution");
            else
            {
                std::stringstream msg;
                msg << "Incorrect output: " << x << " " << y;
                QuitWith(WA, msg.str());
            }
        }
        case '?':
        {
            if (f)
                std::cout << f1(x, y) << std::endl;
            else
                std::cout << f2(x, y) << std::endl;
            break;
        }
        default:
        {
            std::stringstream msg;
            msg << "Incorrect input command: " << requestType;
            QuitWith(PE, msg.str());
        }
        break;
        }
        Out.Match('\n');
    }
    QuitWith(EF, "Incorrect: Input after EOF");
    std::cerr << "INT\n";
    return 0;
}

File: checkers/228/main.go
package main

import (
	"fmt"
	"math"
	"testlib"
	"testlib/core"
	"testlib/utils"
)

func f1(x, y float64) float64 {
	return x*x + y*y - 0.1*math.Abs(1-x) - 0.1*math.Abs(1-y)
}

func f2(x, y float64) float64 {
	return 20*math.Abs(x-50)*math.Abs(y-25) + 10*(math.Abs(x-10)*math.Abs(y-10)+math.Abs(x-50))
}

func dist(anX, anY, x, y float64) float64 {
	return math.Sqrt(math.Pow(anX-x, 2) + math.Pow(anY-y, 2))
}

func i2b(i_0 int) bool {
	if i_0 == 0 {
		return false
	} else {
		return true
	}
}

func createMatrix0(n int, m int) [][]int {
	matrix := make([][]int, n)
	for i := 0; i < n; i++ {
		matrix[i] = make([]int, m)
		for j := 0; j < m; j++ {
			matrix[i][j] = 0
		}
	}
	return matrix
}

func checkValueOutOfSegmen(val int, a int, b int) bool {
	if val < a || val > b {
		return true
	} else {
		return false
	}
}

func main() {
	defer func() {
		_ = testlib.GetStream(testlib.TaskOutput).Close()
		_ = testlib.GetStream(testlib.InputFromUser).Close()
	}()
	inputStream := testlib.GetReader(testlib.TaskInput)
	//outputStream := testlib.GetReader(testlib.TaskOutput)
	inputFromUserStream := testlib.GetReader(testlib.InputFromUser)
	outputToUserStream := testlib.GetWriter(testlib.OutputToUser)

	// n := core.ReadInt(inputStream)
	var eps float64 = 0.001
	f := core.ReadInt(inputStream)
	k := 0
	// ans := make([]float64, 0)
	var x, y float64
	cnt := 0
	var requestType byte
	for {
		if flag, err := utils.IsStreamFinish(inputFromUserStream, inputFromUserStream); flag {
			if err != nil {
				testlib.WriteAnswer(err.Error(), testlib.EF)
				return
			}
			break
		}
		k++
		if cnt > 100000 {
			testlib.WriteAnswer(fmt.Sprintf("Too many requests"), testlib.WA)
			return
		}
		requestType = core.ReadByte(inputFromUserStream)
		x = core.ReadFloat(inputFromUserStream)
		y = core.ReadFloat(inputFromUserStream)
		switch requestType {
		case byte('!'):
			if i2b(f) {
				if math.Abs(x+0.05) < eps && math.Abs(y+0.05) < eps {
					testlib.WriteAnswer(fmt.Sprintf("OK"), testlib.OK)
					return
				} else {
					testlib.WriteAnswer(fmt.Sprintf("Incorrect output: %.3f %.3f", x, y), testlib.WA)
					return
				}
			} else if math.Abs(50-x) < eps && math.Abs(10-y) < eps {
				testlib.WriteAnswer(fmt.Sprintf("OK"), testlib.OK)
				return
			} else {
				testlib.WriteAnswer(fmt.Sprintf("Incorrect output: %.3f %.3f", x, y), testlib.WA)
				return
			}
		case byte('?'):
			if i2b(f) {
				core.WriteStringToWriter(outputToUserStream, fmt.Sprintf("%.20f\n", f1(x, y)))
				core.FlushWriter(outputToUserStream)
			} else {
				core.WriteStringToWriter(outputToUserStream, fmt.Sprintf("%.20f\n", f2(x, y)))
				core.FlushWriter(outputToUserStream)
			}
			break
		default:
			testlib.WriteAnswer(fmt.Sprintf("Incorrect input command: %s", string(requestType)), testlib.PE)
			return
		}
	}
	testlib.WriteAnswer(fmt.Sprintf("Incorrect: Input after EOF"), testlib.EF)
	return
}
